src/main.d

```diff
@@ -607,20 +607,41 @@ int main(string[] args)
 		itemDb = new ItemDatabase(cfg.databaseFilePathDryRun);
 	}
 	
+	// What are the permission that have been set for the application?
+	// These are relevant for:
+	// - The ~/OneDrive parent folder or 'sync_dir' configured item
+	// - Any new folder created under ~/OneDrive or 'sync_dir'
+	// - Any new file created under ~/OneDrive or 'sync_dir'
+	// valid permissions are 000 -> 777 - anything else is invalid
+	if ((cfg.getValueLong("sync_dir_permissions") > 777) || (cfg.getValueLong("sync_file_permissions") > 777)) {
```

Comment:
I think we allow values <0 here. The ValueLong uses `to!long` for conversion and (but I am not sure!) I guess that this accepts negative numbers, too?

Reply:
Yes .. good catch .. need to trap for negative as well